OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -109,8 +119,20 @@ public AssetBrowserLogic(Widget widget, Action onExit, World world)
 			}
 
 			filenameInput = panel.Get<TextFieldWidget>("FILENAME_INPUT");
+			filenameInput.TakeKeyboardFocus();
 			filenameInput.OnTextEdited = () => ApplyFilter(filenameInput.Text);
-			filenameInput.OnEscKey = filenameInput.YieldKeyboardFocus;
+			filenameInput.OnEscKey = () =>
+			{
+				if (filenameInput.Text.Length == 0)
+					closeButton.OnClick();
+				else
+				{
+					filenameInput.Text = null;
```

Comment:
This seems dangerous to me. In general you want to use "" over null when it comes to UI text, and it looks to me like this could break if this handler runs twice.